docs/reference/async-processing.mdx

```diff
@@ -0,0 +1,116 @@
+---
+title: Async Processing
+---
+import {Icons} from '/src/components/icons'
+
+:::info
+
+This page covers the Responsive Async Processor: its benefits, its limitations, how to use it, and how to tune it.
+
+The Async Processor is available with Responsive client **v0.24.0** onward.
+:::
+
+## Why use the Async Processor
+Async processing allows Kafka Streams applications to concurrently process records within a single task, thus changing the upper limit of parallelization from the level of a Kafka partition to the level of a record key. 
+
+It works by giving each StreamThread a dedicated thread pool which it uses to process the records passing through an async processor. Records of the same key are still processed in order and all the same consistency and correctness guarantees of Kafka Streams are maintained, including exactly-once semantics.
+
+While async processing can help improve performance of any application with a large keyspace, it was built to shine when processors include a long-running step such as remote state store requests or generic RPCs. The classic solution for achieving high throughput when you are bottlenecked by network latency is to increase the parallelism of the calls to the target system, which is exactly what the async processor does for Kafka Streams. 
```

Comment:
I know this was my own wording, but I'm wondering if we should say "large enough" keyspace, or hedge it even further? I'm worried this sounds like it won't help at all unless you have a huge number of unique keys, which isn't exactly true -- to be fair I haven't tested very much with smaller keyspaces, but it should still improve things as long as you have eg more keys than async threads, which should always be the case and would be a very small keyspace indeed

Does that make sense? Not exactly sure how to reword this..I guess we could just remove the "with a large keyspace" part of the sentence altogether...thoughts?

Reply:
Yes. Let's drop it and focus on the use case it was built for. I'll reword.